code/modules/client/preference_setup/general/04_equipment.dm

```diff
@@ -35,14 +35,14 @@
 	var/undies = get_undies()
 	var/gender_socks = get_gender_socks()
 	if(!get_key_by_value(undies, pref.underwear))
-		pref.underwear = undies[undies[1]]
+		pref.underwear = null
 	if(!get_key_by_value(undershirt_t, pref.undershirt))
 		pref.undershirt = undershirt_t[undershirt_t[1]]
 	if(!get_key_by_value(gender_socks, pref.socks))
-		pref.socks = gender_socks[gender_socks[1]]
-
+		pref.socks = null

```

Comment:
This most likely isn't the best place to put it. I'm uncertain how safe `null`-ing this is. It might be a better idea to bind to where-ever you swap genders:
```dm
	// Whatever setting of gender entails goes here.
	var/gender_socks = get_gender_socks()
	pref.socks = gender_socks[gender_socks[1]]
	var/undies = get_undies()
	pref.underwear = undies[undies[1]]
```